core/vm/contracts.go

```diff
@@ -358,3 +375,49 @@ func (c *bn256Pairing) Run(input []byte) ([]byte, error) {
 	}
 	return false32Byte, nil
 }
+
+type blake2F struct{}
+
+func (c *blake2F) RequiredGas(input []byte) uint64 {
+	return 1 //TODO: implement
```

Comment:
This still needs to be implemented but we need to decide first what's the strategy.

Reply:
Benchmarked on my Mac:
```
BenchmarkPrecompiledEcrecover/-Gas=3000-8                  10000            180309 ns/op
```

```
BenchmarkPrecompiledSha256/128-Gas=108-8                 2000000               614 ns/op
```

```
BenchmarkPrecompiledBlake2F/testVectors2bX_0-Gas=1-8             3000000               382 ns/op
BenchmarkPrecompiledBlake2F/testVectors2bX_3-Gas=1-8             3000000               383 ns/op
BenchmarkPrecompiledBlake2F/testVectors2bX_70-Gas=1-8            5000000               382 ns/op
BenchmarkPrecompiledBlake2F/testVectors2bX_140-Gas=1-8           3000000               374 ns/op
BenchmarkPrecompiledBlake2F/testVectors2bX_230-Gas=1-8           5000000               379 ns/op
BenchmarkPrecompiledBlake2F/testVectors2bX_300-Gas=1-8           5000000               387 ns/op
BenchmarkPrecompiledBlake2F/testVectors2bX_370-Gas=1-8           5000000               462 ns/op
BenchmarkPrecompiledBlake2F/testVectors2bX_440-Gas=1-8           5000000               477 ns/op
BenchmarkPrecompiledBlake2F/testVectors2bX_510-Gas=1-8           3000000               421 ns/op
BenchmarkPrecompiledBlake2F/testVectors2bX_580-Gas=1-8           5000000               376 ns/op
```

Reply:
Continuing the thread in the [EIP PR](https://github.com/keep-network/EIPs/pull/2/files#r295576690)